parsl/providers/kubernetes/kube.py

```diff
@@ -181,6 +180,93 @@ def _status(self):
         return jobs_ids
         # do something to get the deployment's status
 
+    def _create_pod(self,
+                    image,
+                    pod_name,
+                    job_name,
+                    port=80,
+                    cmd_string=None,
+                    volumes=[]):
+        """ Create a kubernetes non-deployment pod for the job.
+        Args:
+              - image (string) : Docker image to launch
+              - pod_name (string) : Name of the pod
+              - job_name (string) : App label
+        KWargs:
+             - port (integer) : Container port
+        Returns:
+              - None
+        """
+
+        security_context = None
+        if self.user_id and self.group_id:
+            security_context = client.V1SecurityContext(run_as_group=self.group_id,
+                                                        run_as_user=self.user_id,
+                                                        run_as_non_root=self.run_as_non_root)
+
+        # Create the enviornment variables and command to initiate IPP
+        environment_vars = client.V1EnvVar(name="TEST", value="SOME DATA")
+
+        launch_args = ["-c", "{0};".format(cmd_string)]
+
+        volume_mounts = []
+        # Create mount paths for the volumes
+        for volume in volumes:
+            volume_mounts.append(client.V1VolumeMount(mount_path=volume[1],
+                                                      name=volume[0]))
+        # Configureate Pod template container
+        container = None
+        if security_context:
```

Comment:
I think the only purpose of this is to supply or not supply the `security_context` parameter. It looks (from the generated python code of `V1Container` in https://github.com/kubernetes-client/python/blob/master/kubernetes/client/models/v1_container.py) that it is fine to pass in `None` for `security_context` and so this `if`-statement could collapse into a single code path.

Reply:
collapsed